File: 프로그래머스/숫자의 표현.py
def solution(n):
    # 자기 자신 경우의 수 미리 포함
    answer = 1
    # 절반 언저리 부터 시작
    k = n // 2 + 1
    flag = True
    
    while(flag):
        ssum = 0
        kk = k
        # 연속된 수의 합이 n보다 크면 중단하고 1작은수로 다시 누적합 시작
        while(ssum <= n):
            ssum += kk
            # 정답 발견하면 그만 계산하고 다음 케이스로 이동
            if ssum == n:
                answer += 1
                break
            # 끝에 1까지 찍으면 더이상 경우의 수 존재 안하므로 전체 종료
            if kk == 1:
                flag = False
                break
            kk -= 1
        k -= 1
    
    return answer

File: 프로그래머스/숫자의 표현.java
class Solution {
    public int solution(int n) {
        // 자기 자신의 경우의 수 +1
        int count = 1;
        // 연속된 수이기 때문에 그 수의 절반 언저리부터 체크하면 된다
        for (int i = n/2 + 1; i > 0; i--) {
            int sum = 0;
            // 1씩 감소하면서 합해주고 총합이 정답이거나 정답보다 커지면 탈출
            for (int j = i; j > 0; j--) {
                sum += j;
                if (sum == n) {
                    count++;
                    break;
                }
                if (sum > n)
                    break;
            }
        }
        return count;
    }
}